mdaviz/chartview.py

```diff
@@ -117,3 +123,203 @@ def setLeftAxisUnits(self, text):
+        self.figure = Figure()
+        self.canvas = FigureCanvas(self.figure)
+        # Adjust layout to fit the size and position of the axes within the figure
+        # self.figure.tight_layout(pad=2.0, w_pad=0.5, h_pad=0.5)
+        # Create the main plot subplot (flexible size)
+        self.main_axes = self.figure.add_subplot(111)
+        # Adjust margins
+        self.figure.subplots_adjust(bottom=0.16, top=0.92, right=0.92)
+
+        # Create the info panel subplot (fixed size)
+        self.info_panel_axes = self.figure.add_axes([0.1, 0.01, 0.8, 0.1])
+        self.info_panel_axes.set_frame_on(False)
+        self.info_panel_axes.get_xaxis().set_visible(False)
+        self.info_panel_axes.get_yaxis().set_visible(False)
+
+        # Create the navigation toolbar
+        self.toolbar = NavigationToolbar(self.canvas, self)
+
+        # Use a QVBoxLayout for stacking the toolbar and canvas vertically
+        layout = QtWidgets.QVBoxLayout(self)
+        layout.addWidget(self.toolbar)
+        layout.addWidget(self.canvas)
+        # Apply the QVBoxLayout to the ChartViewMpl widget
+        self.setLayout(layout)
+
+        # Connect the click event to a handler
+        self.cid = self.figure.canvas.mpl_connect("button_press_event", self.onclick)
+        self.cursors = {
+            1: False,
+            "pos1": None,
+            2: False,
+            "pos2": None,
+        }
+
+        # Plot configuration
+        config = {
+            "title": self.setPlotTitle,
+            "y": self.setLeftAxisText,
+            "x": self.setBottomAxisText,
+        }
+        for k, func in config.items():
+            func(kwargs.get(k))
+
+    def plot(self, *args, **kwargs):
+        # Plot data on the axes
+        self.main_axes.plot(*args, **kwargs)
+        self.main_axes.legend()
+        self.main_axes.grid(True, color="#cccccc", linestyle="-", linewidth=0.5)
+        self.update_info_panel()
+        self.canvas.draw()
+
+    def setPlotTitle(self, text):
+        self.main_axes.set_title(text, fontsize=FONTSIZE, y=1.03)
+
+    def setBottomAxisText(self, text):
+        self.main_axes.set_xlabel(text, fontsize=FONTSIZE, labelpad=10)
+
+    def setLeftAxisText(self, text):
+        self.main_axes.set_ylabel(text, fontsize=FONTSIZE, labelpad=20)
+
+    def clearPlot(self):
+        self.main_axes.clear()
+        self.cursors[1] = False
+        self.cursors[2] = False
+        self.info_panel_axes.clear()
+        self.canvas.draw()
+
+    # Additional methods:
+
+    # TODO: use a dictionary to track the cursors
+    # TODO: allow a blue cursor to be placed without the presence of a red cursor
+
+    def onclick(self, event):
+        # Check if the click was in the main_axes
+        if event.inaxes is self.main_axes:
+            # Middle click for red cursor
+            if event.button == 2:
+                if self.cursors[1]:
+                    self.cursors[1].remove()  # Remove existing red cursor
```

Comment:
If `self.cursors[1]` is a boolean, how does it have a `remove()` method?

Reply:
It's either `False` (default) or an artist structure from MPL.  Might be better to default as `None` (which implies some logic changes that test for `None`).